tests/lib/common.py

```diff
@@ -49,18 +49,30 @@ def decode_wrapper(i):
 
 
 def wait_for_result(func, *args, matcher=simple_matcher(True), attempts=20,
-                    interval=5, decode=decode_wrapper, **kwargs):
+                    interval=5, decode=decode_wrapper, ignore_exceptions=True,
+                    **kwargs):
     """Runs `func` with `args` until `matcher(out)` returns true or timesout
 
     Returns the matching result, or raises an exception.
     """
 
-    for i in range(attempts):
-        out = func(*args, **kwargs)
-        if decode:
-            out = decode(out)
-        if matcher(out):
-            return out
+    i = 1
+    while i < (attempts + 1):
+        logger.info(f"Attempt {{ i }} | {{ attempts-i }} left")
```

Comment:
This will produce incorrect notifcations, for example for attempts=3 it will print pairs:
1 / 2
2 / 1
3 / 0
Which is definitely what we do not expect.

Reply:
I don't get what you mean

```
In [15]: i = 1

In [16]: attempts = 3

In [17]: while i < (attempts+1):
    ...:     print("Attempt %d | %d left" % (i, attempts - i))
    ...:     i = i + 1
    ...: 
Attempt 1 | 2 left
Attempt 2 | 1 left
Attempt 3 | 0 left
```

Reply:
@brunoleon Okay, this is useless information, the useful format is `number of try / total number` or just number of tries left.

Reply:
You either report number of tries from total, or just how many tries left, because `Attempt X | Y left` is not clear what you mean. It can be read like `Attempt X from Y left`.

Reply:
Updated according to your request